src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/Patch/FhirPatchDeleteTests.cs

```diff
@@ -19,9 +21,9 @@ public void GivenAFhirPatchDeleteRequest_WhenDeletingPrimitive_ThenPrimitiveShou
         {
             var patchParam = new Parameters().AddDeletePatchParameter("Patient.birthDate");
 
-            Patient patchedPatientResource = (Patient)new FhirPathPatchBuilder(new Patient { BirthDate = "1920-01-01" }, patchParam).Apply();
+            Patient patchedPatientResource = new FhirPathPatchBuilder(new Patient { BirthDate = "1920-01-01" }, patchParam).Apply() as Patient;
 
-            Assert.True(patchedPatientResource.Matches(new Patient()));
+            Assert.Equal(patchedPatientResource.ToJson(), new Patient().ToJson());
```

Comment:
Just curious as to the change from Matches to a string compare on the serialized Json.  Was the Matches method failing?

Reply:
Matches is quite primitive - it doesn't compare all the elements. 👎🏽 So a lot of tests were testing nothing. 

https://github.com/FirelyTeam/firely-net-sdk/blob/ac40e368b8c09e266cb727edb89c9448b65e178c/src/Hl7.Fhir.Core/Model/Generated/Patient.cs#L245